# [BICF](http://www.utsouthwestern.edu/labs/bioinformatics/) Genome-Wide Association Studies Nano Course

Even wonder how to identify genetic biomarkers? Genome-Wide Association Studies (GWAS) use statistical methods to identify the variants that are associated to traits by identifying differences in allele frequency in binary traits represented in 2 populations (case vs control) or in a quantitative trait within a population, such as height. This class will cover the good, the bad and the ugly of GWAS analysis including (i) study design, (ii) confounding factors including ethnicity and genetic linkage, (iii) population structure, (iv) how to determine relatedness between individuals and (v) how to perform this analysis for your own research. This is a hand-on workshop, which will include working in a linux environment.


**Here is a machine image that we will use:**
[Machine Image](https://cloud.biohpc.swmed.edu/index.php/s/mjofGJL5JMsM9z4)

| Time  | Topic | Instructor|
| ------------- | ------------- | ------------- |
| | Thursday, April 27, 2017 | |
| 9 - 10 a.m. | [Overview of Genome-Wide Association Study Analysis](img/rationalegwasxing.pdf) | Chao Xing |
| 10 - 12 a.m.  | Examining GWAS Data Using Haploview <br>[References](img/2014xingiovs7073.pdf) | |
| 1:30 - 2:30 p.m.|[Study Design, Examination of Population Structure and Genetic Relationships](img/gwas_zhang_lecture.pdf)| He Zhang |
| 2:30 - 4:30 p.m. | [Scripting Workshop](img/gwas_zhang_tutorial.pdf) <br> [Data](https://www.dropbox.com/s/7k4a96i9mxzhpz9/course2.tar.bz2?dl=0) | |
| | April 28, 2017 | |
| 9 - 10 a.m. | [Haplotype Analysis and Genotype Imputation](img/phasing_and_imputation_v2.pdf) | Xiaowei Zhan |
| 10 - 12 a.m.  | [ Phasing Workshop](phasing.html) <br> [Imputation Workshop](imputation.html) | Xiaowei Zhan|
| 1:30 - 2:30 p.m.|[Disease models for quantitative or binary traits](kozlitina_gwas_analysis_04282017.pdf)| Julia Kozlitina |
| 2:30 - 4:30 p.m. | [GWAS Workflow Development using PLINK](workshop_04282017.pdf) | |
