<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns:mv="http://macVmlSchemaUri" xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta name=Title content="">
<meta name=Keywords content="">
<meta http-equiv=Content-Type content="text/html; charset=macintosh">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 14">
<meta name=Originator content="Microsoft Word 14">
<link rel=File-List href="course_16S_files/filelist.xml">
<!--[if gte mso 9]><xml>
 <o:DocumentProperties>
  <o:Author>Brandi Cantarel</o:Author>
  <o:LastAuthor>Brandi Cantarel</o:LastAuthor>
  <o:Revision>2</o:Revision>
  <o:TotalTime>83</o:TotalTime>
  <o:Created>2015-10-18T23:40:00Z</o:Created>
  <o:LastSaved>2015-10-18T23:40:00Z</o:LastSaved>
  <o:Pages>2</o:Pages>
  <o:Words>1606</o:Words>
  <o:Characters>9155</o:Characters>
  <o:Company>UMIGS</o:Company>
  <o:Lines>76</o:Lines>
  <o:Paragraphs>21</o:Paragraphs>
  <o:CharactersWithSpaces>10740</o:CharactersWithSpaces>
  <o:Version>14.0</o:Version>
 </o:DocumentProperties>
 <o:OfficeDocumentSettings>
  <o:AllowPNG/>
 </o:OfficeDocumentSettings>
</xml><![endif]-->
<link rel=themeData href="course_16S_files/themedata.xml">
<!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:Zoom>FullPage</w:Zoom>
  <w:TrackMoves/>
  <w:TrackFormatting/>
  <w:PunctuationKerning/>
  <w:DrawingGridHorizontalSpacing>18 pt</w:DrawingGridHorizontalSpacing>
  <w:DrawingGridVerticalSpacing>18 pt</w:DrawingGridVerticalSpacing>
  <w:DisplayHorizontalDrawingGridEvery>0</w:DisplayHorizontalDrawingGridEvery>
  <w:DisplayVerticalDrawingGridEvery>0</w:DisplayVerticalDrawingGridEvery>
  <w:ValidateAgainstSchemas/>
  <w:SaveIfXMLInvalid>false</w:SaveIfXMLInvalid>
  <w:IgnoreMixedContent>false</w:IgnoreMixedContent>
  <w:AlwaysShowPlaceholderText>false</w:AlwaysShowPlaceholderText>
  <w:DoNotPromoteQF/>
  <w:LidThemeOther>EN-US</w:LidThemeOther>
  <w:LidThemeAsian>JA</w:LidThemeAsian>
  <w:LidThemeComplexScript>X-NONE</w:LidThemeComplexScript>
  <w:Compatibility>
   <w:BreakWrappedTables/>
   <w:SnapToGridInCell/>
   <w:WrapTextWithPunct/>
   <w:UseAsianBreakRules/>
   <w:DontGrowAutofit/>
   <w:SplitPgBreakAndParaMark/>
   <w:EnableOpenTypeKerning/>
   <w:DontFlipMirrorIndents/>
   <w:OverrideTableStyleHps/>
   <w:UseFELayout/>
  </w:Compatibility>
  <m:mathPr>
   <m:mathFont m:val="Cambria Math"/>
   <m:brkBin m:val="before"/>
   <m:brkBinSub m:val="&#45;-"/>
   <m:smallFrac/>
   <m:dispDef/>
   <m:lMargin m:val="0"/>
   <m:rMargin m:val="0"/>
   <m:defJc m:val="centerGroup"/>
   <m:wrapRight/>
   <m:intLim m:val="subSup"/>
   <m:naryLim m:val="subSup"/>
  </m:mathPr></w:WordDocument>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:LatentStyles DefLockedState="false" DefUnhideWhenUsed="true"
  DefSemiHidden="true" DefQFormat="false" DefPriority="99"
  LatentStyleCount="276">
  <w:LsdException Locked="false" Priority="0" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Normal"/>
  <w:LsdException Locked="false" Priority="9" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="heading 1"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 2"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 3"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 4"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 5"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 6"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 7"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 8"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 9"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 1"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 2"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 3"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 4"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 5"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 6"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 7"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 8"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 9"/>
  <w:LsdException Locked="false" Priority="35" QFormat="true" Name="caption"/>
  <w:LsdException Locked="false" Priority="10" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Title"/>
  <w:LsdException Locked="false" Priority="1" Name="Default Paragraph Font"/>
  <w:LsdException Locked="false" Priority="11" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Subtitle"/>
  <w:LsdException Locked="false" Priority="22" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Strong"/>
  <w:LsdException Locked="false" Priority="20" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Emphasis"/>
  <w:LsdException Locked="false" Priority="59" SemiHidden="false"
   UnhideWhenUsed="false" Name="Table Grid"/>
  <w:LsdException Locked="false" UnhideWhenUsed="false" Name="Placeholder Text"/>
  <w:LsdException Locked="false" Priority="1" UnhideWhenUsed="false"
   QFormat="true" Name="No Spacing"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 1"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 1"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 1"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 1"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 1"/>
  <w:LsdException Locked="false" UnhideWhenUsed="false" Name="Revision"/>
  <w:LsdException Locked="false" Priority="34" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="List Paragraph"/>
  <w:LsdException Locked="false" Priority="29" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Quote"/>
  <w:LsdException Locked="false" Priority="30" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Intense Quote"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 1"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 1"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 1"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 1"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 1"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 1"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 2"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 2"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 2"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 2"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 2"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 2"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 2"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 2"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 3"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 3"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 3"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 3"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 3"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 3"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 3"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 3"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 4"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 4"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 4"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 4"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 4"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 4"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 4"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 4"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 5"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 5"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 5"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 5"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 5"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 5"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 5"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 5"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 6"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 6"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 6"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 6"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 6"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 6"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 6"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 6"/>
  <w:LsdException Locked="false" Priority="19" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Subtle Emphasis"/>
  <w:LsdException Locked="false" Priority="21" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Intense Emphasis"/>
  <w:LsdException Locked="false" Priority="31" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Subtle Reference"/>
  <w:LsdException Locked="false" Priority="32" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Intense Reference"/>
  <w:LsdException Locked="false" Priority="33" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Book Title"/>
  <w:LsdException Locked="false" Priority="37" Name="Bibliography"/>
  <w:LsdException Locked="false" Priority="39" QFormat="true" Name="TOC Heading"/>
 </w:LatentStyles>
</xml><![endif]-->
<style>
<!--
 /* Font Definitions */
@font-face
	{font-family:Times;
	panose-1:2 0 5 0 0 0 0 0 0 0;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:"\FF2D\FF33 \660E\671D";
	panose-1:0 0 0 0 0 0 0 0 0 0;
	mso-font-charset:128;
	mso-generic-font-family:roman;
	mso-font-format:other;
	mso-font-pitch:fixed;
	mso-font-signature:1 134676480 16 0 131072 0;}
@font-face
	{font-family:"\FF2D\FF33 \660E\671D";
	panose-1:0 0 0 0 0 0 0 0 0 0;
	mso-font-charset:128;
	mso-generic-font-family:roman;
	mso-font-format:other;
	mso-font-pitch:fixed;
	mso-font-signature:1 134676480 16 0 131072 0;}
@font-face
	{font-family:Cambria;
	panose-1:2 4 5 3 5 4 6 3 2 4;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:Georgia;
	panose-1:2 4 5 2 5 4 5 2 3 3;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:Monaco;
	panose-1:2 0 5 0 0 0 0 0 0 0;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
 /* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:Cambria;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:"\FF2D\FF33 \660E\671D";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
h2
	{mso-style-priority:9;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-link:"Heading 2 Char";
	mso-margin-top-alt:auto;
	margin-right:0in;
	mso-margin-bottom-alt:auto;
	margin-left:0in;
	mso-pagination:widow-orphan;
	mso-outline-level:2;
	font-size:18.0pt;
	font-family:Times;}
a:link, span.MsoHyperlink
	{mso-style-priority:99;
	color:blue;
	mso-themecolor:hyperlink;
	text-decoration:underline;
	text-underline:single;}
a:visited, span.MsoHyperlinkFollowed
	{mso-style-noshow:yes;
	mso-style-priority:99;
	color:purple;
	mso-themecolor:followedhyperlink;
	text-decoration:underline;
	text-underline:single;}
pre
	{mso-style-noshow:yes;
	mso-style-priority:99;
	mso-style-link:"HTML Preformatted Char";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:Courier;
	mso-fareast-font-family:"\FF2D\FF33 \660E\671D";
	mso-fareast-theme-font:minor-fareast;
	mso-bidi-font-family:Courier;}
span.Heading2Char
	{mso-style-name:"Heading 2 Char";
	mso-style-priority:9;
	mso-style-unhide:no;
	mso-style-locked:yes;
	mso-style-link:"Heading 2";
	mso-ansi-font-size:18.0pt;
	mso-bidi-font-size:18.0pt;
	font-family:Times;
	mso-ascii-font-family:Times;
	mso-hansi-font-family:Times;
	font-weight:bold;}
span.mw-headline
	{mso-style-name:mw-headline;
	mso-style-unhide:no;}
span.HTMLPreformattedChar
	{mso-style-name:"HTML Preformatted Char";
	mso-style-noshow:yes;
	mso-style-priority:99;
	mso-style-unhide:no;
	mso-style-locked:yes;
	mso-style-link:"HTML Preformatted";
	mso-ansi-font-size:10.0pt;
	mso-bidi-font-size:10.0pt;
	font-family:Courier;
	mso-ascii-font-family:Courier;
	mso-hansi-font-family:Courier;
	mso-bidi-font-family:Courier;}
span.apple-converted-space
	{mso-style-name:apple-converted-space;
	mso-style-unhide:no;}
.MsoChpDefault
	{mso-style-type:export-only;
	mso-default-props:yes;
	font-family:Cambria;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:"\FF2D\FF33 \660E\671D";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
@page WordSection1
	{size:8.5in 11.0in;
	margin:1.0in 1.25in 1.0in 1.25in;
	mso-header-margin:.5in;
	mso-footer-margin:.5in;
	mso-paper-source:0;}
div.WordSection1
	{page:WordSection1;}
-->
</style>
<!--[if gte mso 10]>
<style>
 /* Style Definitions */
table.MsoNormalTable
	{mso-style-name:"Table Normal";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-priority:99;
	mso-style-parent:"";
	mso-padding-alt:0in 5.4pt 0in 5.4pt;
	mso-para-margin:0in;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:Cambria;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;}
</style>
<![endif]--><!--[if gte mso 9]><xml>
 <o:shapedefaults v:ext="edit" spidmax="1026"/>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <o:shapelayout v:ext="edit">
  <o:idmap v:ext="edit" data="1"/>
 </o:shapelayout></xml><![endif]-->
</head>

<body lang=EN-US link=blue vlink=purple style='tab-interval:.5in'>

<div class=WordSection1>

<p class=MsoNormal><span style='font-family:Helvetica'>This exercise is adapted
from the Mothur MiSeq SOP<o:p></o:p></span></p>

<p class=MsoNormal><a href="http://www.mothur.org/wiki/MiSeq_SOP"><span
style='font-family:Helvetica'>http://www.mothur.org/wiki/MiSeq_SOP</span></a><span
style='font-family:Helvetica'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><i style='mso-bidi-font-style:normal'><span
style='font-size:10.0pt;font-family:Courier;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";color:#252525;background:white'>Kozich
JJ, Westcott SL, Baxter NT, Highlander SK, Schloss PD. (2013): Development of a
dual-index sequencing strategy and curation pipeline for analyzing amplicon
sequence data on the MiSeq Illumina sequencing platform. Applied and
Environmental Microbiology. 79(17):5112-20.</span></i><i style='mso-bidi-font-style:
normal'><span style='font-size:10.0pt;font-family:Courier;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></i></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<div style='mso-element:para-border-div;border:none;border-bottom:solid #AAAAAA 1.0pt;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in 0in 0in 0in;background:
white'>

<p class=MsoNormal style='margin-top:12.0pt;margin-right:0in;margin-bottom:
3.0pt;margin-left:0in;mso-outline-level:2;background:white;border:none;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in;mso-padding-alt:0in 0in 0in 0in'><span
style='font-size:15.0pt;font-family:Georgia;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";color:black'>Reducing sequencing and PCR
errors<o:p></o:p></span></p>

</div>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>In this exercise we are
going to examine the microbiome from the pig intestinal track from one
donor.<span style="mso-spacerun:yes">&nbsp; </span>These sequences were
generated by 454 downloaded from SRA.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><i
style='mso-bidi-font-style:normal'><span style='font-size:10.0pt;font-family:
Courier;mso-bidi-font-family:Courier;color:black'>Looft T, Allen HK, Cantarel
BL, Levine UY, Bayles DO, Alt DP, Henrissat B, Stanton TB. Bacteria, phages and
pigs: the effects of in-feed antibiotics on the<span
style="mso-spacerun:yes">&nbsp; </span>microbiome at different gut locations.
ISME J. 2014 Aug;8(8):1566-76. doi: 10.1038/ismej.2014.12. Epub 2014 Feb 13.
PubMed PMID: 24522263.<o:p></o:p></span></i></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>First lets examine the
sequence data<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>summary.seqs(fasta=swine_gut.fasta)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Now lets trim sequences
based on quality, remove any large, small, ambiguious or reads with lots of
homopolymers<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>trim.seqs(fasta=swine_gut.fasta,qfile=swine_gut.qual,qaverage=25,maxambig=0,maxhomop=8,minlength=200,maxlength=800,processors=2)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Now we are going to
reduce the number of reads for analysis. Don’t worry we are “keeping” these
sequences for abundance purposes, just not for analysis.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>unique.seqs(fasta=swine_gut.trim.fasta)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<div style='mso-element:para-border-div;border:none;border-bottom:solid #AAAAAA 1.0pt;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in 0in 0in 0in;background:
white'>

<h2 style='margin-top:12.0pt;margin-right:0in;margin-bottom:3.0pt;margin-left:
0in;background:white;border:none;mso-border-bottom-alt:solid #AAAAAA .75pt;
padding:0in;mso-padding-alt:0in 0in 0in 0in'><span class=mw-headline><span
style='font-size:15.0pt;font-family:Georgia;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";color:black;font-weight:normal'>Processing
improved sequences</span></span><span style='font-size:15.0pt;font-family:Georgia;
mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman";
color:black;font-weight:normal'><o:p></o:p></span></h2>

</div>

<p class=MsoNormal><span style='font-family:Helvetica'>Normally, we can limit
the alignment to the variable region of the 16S gene based on the primers.<span
style="mso-spacerun:yes">&nbsp; </span>First thing we would do is to reduce the
16S reference alignment to this region.<span style="mso-spacerun:yes">&nbsp;
</span>But this step is very long, so we are going to use the precut alignment.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>#pcr.seqs(fasta=silva.nr_v119.align,
start=1044, end=6500,keepdots=F)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Next align sequences to
template 16S rRNA gene.<span style="mso-spacerun:yes">&nbsp; </span>We can
remove any sequences that don’t align to the expected PCR region.<span
style="mso-spacerun:yes">&nbsp;&nbsp; </span>The alignment step can also be
very long &#8211; so you should probably skip this step too.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>#align.seqs(fasta=swine_gut.trim.unique.fasta,template=silva.nr_v119.pcr.align,flip=t,processors=2)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Summarize results so far<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>summary.seqs(fasta=swine_gut.trim.unique.align,
name=swine_gut.trim.names)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Next we want to create a
file to keep track of the count of our representative sequences in each
sample.<span style="mso-spacerun:yes">&nbsp; </span>Since we skipped some the
demultiplexing steps, well have to generate the read to same file (group
file).<span style="mso-spacerun:yes">&nbsp; </span>Then create a count table.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>system(perl
make_group_names.pl swine_gut.trim.names)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>count.seqs(name=swine_gut.trim.names,
group=swine_gut.trim.groups)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Now we want to “clean”
up our alignment, determine the sequences that cover most of the PCR region,
cut the alignment to that region and remove those that don’t span the whole
region<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>screen.seqs(fasta=swine_gut.trim.unique.align,
count=swine_gut.trim.count_table,start=2,optimize=end, criteria=85,
processors=2)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>filter.seqs(fasta=swine_gut.trim.unique.good.align,
vertical=T, trump=., processors=2)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Now that our alignment
is of high quality, let’s remove redundant sequences and cluster sequences with
only 2 mismatches (these could represent sequencing errors)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>unique.seqs(fasta=swine_gut.trim.unique.good.filter.fasta,
count=swine_gut.trim.good.count_table)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>pre.cluster(fasta=swine_gut.trim.unique.good.filter.unique.fasta,count=swine_gut.trim.unique.good.filter.count_table,diffs=2)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>We want to identify and
remove chimera sequences.<span style="mso-spacerun:yes">&nbsp; </span>This step
is really slow, we have already identified these sequences.<span
style="mso-spacerun:yes">&nbsp; </span>So skip this step and remove the
chimeras.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><span
style="mso-spacerun:yes">&nbsp;</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>#</span><span
style='font-family:Courier'>chimera.uchime(fasta=swine_gut.trim.unique.good.filter.unique.precluster.fasta,
reference=silva.gold.align,processors=3) remove.seqs(accnos=swine_gut.trim.unique.good.filter.unique.precluster.ref.uchime.accnos,
fasta=swine_gut.trim.unique.good.filter.unique.precluster.fasta,count=swine_gut.trim.unique.good.filter.unique.precluster.count_table)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<div style='mso-element:para-border-div;border:none;border-bottom:solid #AAAAAA 1.0pt;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in 0in 0in 0in;background:
white'>

<h2 style='margin-top:12.0pt;margin-right:0in;margin-bottom:3.0pt;margin-left:
0in;background:white;border:none;mso-border-bottom-alt:solid #AAAAAA .75pt;
padding:0in;mso-padding-alt:0in 0in 0in 0in'><span class=mw-headline><span
style='font-size:15.0pt;font-family:Georgia;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";color:black;font-weight:normal'>Taxonomic
Classification</span></span><span style='font-size:15.0pt;font-family:Georgia;
mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman";
color:black;font-weight:normal'><o:p></o:p></span></h2>

</div>

<p class=MsoNormal><span style='font-family:Helvetica'>#Classify Sequences into
Taxonomic Bins<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>classify.seqs(fasta=swine_gut.trim.unique.good.filter.unique.precluster.pick.fasta,
template=silva.nr_v119.pcr.align, taxonomy=silva.nr_v119.tax, cutoff=50,
processors=2)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>This command allows you
to remove suspicious sequences, for example if you susptect Cyanobacteria as
representing chloroplast from plant material (which it could in the gut), you
can remove it <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>remove.lineage(fasta=swine_gut.trim.unique.good.filter.unique.precluster.pick.fasta,
count=swine_gut.trim.unique.good.filter.unique.precluster.pick.count_table,
taxonomy=swine_gut.trim.unique.good.filter.unique.precluster.pick.nr_v119.wang.taxonomy,taxon=Cyanobacteria)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>#mothur has a tendency
to make long complicated names, these commands allows you to rename files to
shorter names<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>system(cp
swine_gut.trim.unique.good.filter.unique.precluster.pick.pick.fasta
final.fasta)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>system(cp
swine_gut.trim.unique.good.filter.unique.precluster.pick.pick.count_table
final.count_table)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>system(cp
swine_gut.trim.unique.good.filter.unique.precluster.pick.nr_v119.wang.pick.taxonomy
final.taxonomy)<o:p></o:p></span></p>

<div style='mso-element:para-border-div;border:none;border-bottom:solid #AAAAAA 1.0pt;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in 0in 0in 0in;background:
white'>

<h2 style='margin-top:12.0pt;margin-right:0in;margin-bottom:3.0pt;margin-left:
0in;background:white;border:none;mso-border-bottom-alt:solid #AAAAAA .75pt;
padding:0in;mso-padding-alt:0in 0in 0in 0in'><span class=mw-headline><span
style='font-size:15.0pt;font-family:Georgia;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";color:black;font-weight:normal'>Preparing
for OTU analysis</span></span><span style='font-size:15.0pt;font-family:Georgia;
mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman";
color:black;font-weight:normal'><o:p></o:p></span></h2>

</div>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Create a distance matrix
comparing 16S sequences in your samples.<span style="mso-spacerun:yes">&nbsp;
</span>Then cluster sequences into OTU -- we are using the command
cluster.split to do that as it allows us to cluster sequences according a
taxonomic level like order or family<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>dist.seqs(fasta=final.fasta,cutoff=0.15,
processors=2)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>cluster.split(column=final.dist,
taxonomy=final.taxonomy, count=final.count_table, splitmethod=classify,
taxlevel=4, processors=1)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>To know the taxonomy for
each of our OTUs, get the consensus taxonomy for each OTU<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>classify.otu(list=final.an.unique_list.list,
count=final.count_table, taxonomy=final.taxonomy, label=0.03)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>#Calculate the relative
abundance of taxa<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>phylotype(taxonomy=final.taxonomy,count=final.count_table)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Helvetica'>The number of sequences in your count table
should match the number in your OTU list file.<span
style="mso-spacerun:yes">&nbsp; </span>If time permits create a PERL script to
removes these extra sequences from the count file -- call this new file
final.fix.count_table.<span style="mso-spacerun:yes">&nbsp; </span><i
style='mso-bidi-font-style:normal'>Otherwise use </i></span></b><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica;mso-bidi-font-family:Monaco;color:black'>fix_ct.pl<o:p></o:p></span></i></b></p>

<p class=MsoNormal><span style='font-family:Helvetica;mso-bidi-font-family:
Monaco;color:black'><o:p>&nbsp;</o:p></span></p>

<div style='mso-element:para-border-div;border:none;border-bottom:solid #AAAAAA 1.0pt;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in 0in 0in 0in;background:
white'>

<p class=MsoNormal style='margin-top:12.0pt;margin-right:0in;margin-bottom:
3.0pt;margin-left:0in;mso-outline-level:2;background:white;border:none;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in;mso-padding-alt:0in 0in 0in 0in'><span
style='font-size:15.0pt;font-family:Georgia;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";color:black'>Analysis<o:p></o:p></span></p>

</div>

<p class=MsoNormal><span style='font-size:10.0pt;font-family:Times;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>#Generates collector's
curves<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>#The make.shared creates
a file that represent the number of times that an OTU is observed in multiple
samples<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>make.shared(list=final.an.unique_list.list,count=final.fix.count_table,
label=0.03)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Let's start our analysis
by analyzing the alpha diversity of the samples. First we will generate
collector's curve of the Chao1 richness estimators and the inverse Simpson
diversity index.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>collect.single(shared=final.an.unique_list.shared,
calc=chao-invsimpson, freq=100)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman";color:#252525;
background:white'>This command will generate file ending in *.chao and
*.invsimpson for each sample, which can be plotted in your favorite graphing
software package. It is important to point out that Chao1 is really a measure
of the minimum richness in a community, not the full richness of the community.
Also, there isn't much to do with these plots since it's based on a single
sampling. One method often used to get around this problem is to look at
rarefaction curves describing the number of OTUs observed as a function of
sampling effort.</span><span style='font-family:Times;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>rarefaction.single(shared=final.an.unique_list.shared,
calc=sobs, freq=100)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>#Determine the sequence
count per sample<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>count.groups(shared=final.an.unique_list.shared)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>This will generate files
ending in *.rarefaction, which again can be plotted in your favorite graphing
software package. Alas, rarefaction is not a measure of richness, but a measure
of diversity. If you consider two communities with the same richness, but
different evenness then after sampling a large number of individuals their
rarefaction curves will asymptote to the same value. Since they have different
evennesses the shapes of the curves will differ. Therefore, selecting a number
of individuals to cutoff the rarefaction curve isn't allowing a researcher to
compare samples based on richness, but their diversity. Finally, let's get a
table containing the number of sequences, the sample coverage, the number of
observed OTUs, and the Inverse Simpson diversity estimate using the
summary.single command. To standardize everything, let's randomly select
sequences from each sample 1000 times and calculate the average (note: that if
we set subsample=T, then it would use the size of the smallest library):<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>summary.single(shared=final.an.unique_list.shared,
calc=nseqs-coverage-sobs-invsimpson, subsample=2725)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Subsample the OTUs using
the minimum number of sequences.<span style="mso-spacerun:yes">&nbsp;
</span>Then, let’s determine whether<span style="mso-spacerun:yes">&nbsp;
</span>our data can be partintioned in to separate community types. </span><span
style='font-size:10.0pt;font-family:Times;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>sub.sample(shared=final.an.unique_list.shared,
size=2725)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>get.communitytype(shared=final.an.unique_list.0.03.subsample.shared</span><span
style='font-family:Helvetica'>)</span><span style='font-family:Courier'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman";color:#252525;
background:white'>The unifrac-based metrics are used to assess the similarity
between two communities membership (unifrac.unweighted) and structure
(unifrac.weighted). We will use these metrics and generate PCoA plots to
compare our samples. There are two beta-diversity metrics that one can use -
unweighted and weighted. We will also have mothur subsample the trees 1000
times and report the average.<span style="mso-spacerun:yes">&nbsp; </span></span><span
style='font-family:Helvetica'>In order to do unifract analysis (beta
diversity), we need to determine the distances between sequences, build a
phylogentic tree and then compare samples using unifract.<span
style="mso-spacerun:yes">&nbsp; </span></span><span style='font-family:Helvetica;
mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman";
color:#252525;background:white'>Principal Coordinates (PCoA) uses an
eigenvector-based approach to represent multidimensional data in as few
dimesnsions as possible. Our data is highly dimensional (~9 dimensions).<span
style="mso-spacerun:yes">&nbsp; </span>Tree building is a very slow step.<span
style="mso-spacerun:yes">&nbsp; </span>Please use the pre-generated tree.</span><span
style='font-family:Times;mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:
"Times New Roman"'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman";color:#252525;
background:white'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>#dist.seqs(fasta=final.fasta,cutoff=0.15,
output=lt,processors=4)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>#clearcut(phylip=final.phylip.dist)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>unifrac.unweighted(tree=final.phylip.tre,
count=final.count_table, distance=lt, processors=2, random=F, subsample=2725)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>unifrac.weighted(tree=final.phylip.tre,
count=final.count_table, distance=lt, processors=2, random=F, subsample=2725)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>pcoa(phylip=final.phylip.1.unweighted.ave.dist)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>pcoa(phylip=</span><span
style='font-family:Courier;mso-bidi-font-family:Monaco;color:black'>final.phylip.tre1.weighted.ave.dist</span><span
style='font-family:Courier'>)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>We can do the same
analysis using the </span><span style='font-family:Helvetica;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman";color:#252525;
background:white'>similarity of the membership and structure found in the
various samples</span><span style='font-family:Times;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>dist.shared(shared=final.an.unique_list.shared,
calc=thetayc-jclass, subsample=2725)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>pcoa(phylip=final.an.unique_list.thetayc.0.03.lt.dist)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>nmds(phylip=final.an.unique_list.thetayc.0.03.lt.dist)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Finally, no microbiome
study would be complete without a few heatmaps.<span
style="mso-spacerun:yes">&nbsp; </span>We can visualize those distances as
similarities in a heatmap with the Jaccard and thetayc coefficients.<span
style="mso-spacerun:yes">&nbsp; </span>We can also visualize relative abundance
of each OTU across the 24 samples using the heatmap.bin command and log2
scaling the relative abundance values. Because there are so many OTUs, let's
just look at the top 50 OTUs.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>heatmap.bin(shared=final.an.unique_list.shared,
scale=log2, numotu=50) <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>heatmap.sim(phylip=final.an.unique_list.thetayc.0.03.lt.dist)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>heatmap.sim(phylip=final.an.unique_list.jclass.0.03.lt.dist)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

</div>

</body>

</html>
