---
title: "TCGAbiolinks workshop"
output: html_notebook
---

In this exercise, we will download data from the GDC and analyze it.

Set proxies and working directory
```{r}
Sys.setenv(http_proxy="proxy.swmed.edu:3128")
Sys.setenv(https_proxy="proxy.swmed.edu:3128")
setwd("~/r_nanocourse_2020/")
```

Load TCGAbiolinks package.

```{r}
library(TCGAbiolinks)
```

1. Craft two TCGAbiolinks queries to find tumor and normal BRCA RNA-seq expression count data from GDC (use the `legacy = TRUE` option for integer read counts). 
```{r}
query <- GDCquery(project="TCGA-CHOL", 
                  data.category = "Transcriptome Profiling",
                  sample.type = "Primary solid Tumor", 
                  data.type = "Gene Expression Quantification", 
                  workflow.type = "HTSeq - Counts")

query.2 <- GDCquery(project="TCGA-CHOL", 
                    data.category = "Transcriptome Profiling",
                    sample.type = "Solid Tissue Normal", 
                    data.type = "Gene Expression Quantification", 
                    workflow.type = "HTSeq - Counts")
```




2. We can see how many sampes are in each group by looking in the query results.

```{r}
tumor.patients <- query$results[[1]]$cases
normal.patients <- query.2$results[[1]]$cases

```


3. Download the results of the query.
```{r}
query.3 <- GDCquery(project="TCGA-CHOL", 
                    data.category = "Transcriptome Profiling",
                    #sample.type = "Solid Tissue Normal",      #omit sample.type to get both typs
                    data.type = "Gene Expression Quantification", 
                    workflow.type = "HTSeq - Counts") 
GDCdownload(query.3)

data <- GDCprepare(query.3)
names(data) <- data@rowRanges$external_gene_name

```



4. Using TCGAbiolinks functions:
+ Normalize the data
```{r}
norm.data <- TCGAanalyze_Normalization(data, geneInfo = TCGAbiolinks::geneInfo)
```


+ Perfrom differential expression analysis.
```{r}
deres <- TCGAanalyze_DEA(mat1 = norm.data[,which(colnames(norm.data) %in% normal.patients)], 
                         mat2 = norm.data[,which(colnames(norm.data) %in% tumor.patients)],
                         Cond1type = "normal", Cond2type = "tumor")
```


+ Produce a table of DEG's showing gene names, logFC, FDR, and tumor and normal expression values. 
```{r echo=TRUE}
dataDEGsLevel <- TCGAanalyze_LevelTab(deres,"Tumor","Normal", norm.data[,normal.patients], norm.data[, tumor.patients])
head(dataDEGsLevel)
```

5. Create a heatmap showcasing the expression of the top 40 or so differentially expressed genes (you may wish to calculate logcpm and zscore values for a clearer heatmap).

```{r, eval=FALSE}
library(pheatmap)
dataDEGsLevel <- dataDEGsLevel[order(dataDEGsLevel$FDR, decreasing = FALSE),]
genes.int <- dataDEGsFiltLevel[1:40,1] 
##OR##
genes.int <- row.names(dataDEGsLevel)[1:40]

heat.data <- norm.data[rownames(norm.data) %in% genes.int,]
##or##
heat.data <- norm.data[genes.int,]

heat.data <- scale(heat.data)

pheatmap(heat.data, show_colnames = FALSE, fontsize_row = 6)

```
```{r}
pheatmap(heat.data, show_colnames = FALSE, fontsize_row = 6)
```


