# [About Nanocourses](aboutnanocourses.md)

# Courses Fall 2020

### [Subsribe to course registration mailing list](https://mailchi.mp/utsouthwestern.edu/bicf_nanocourses)

| NGS Sequence Analysis Module | Course Number | Dates |
| ------------- | ------------- | ------------- |
| Genomics Analysis | MB5096 & PDRT5244 | Spring/Summer 2020 |
| Gene Expression Analysis | MB5096 & PDRT5244 | Spring/Summer 2020 |

# Upcoming Courses


# Previous Nanocourses

## [NGS Sequence Analysis Module](https://bicf.pages.biohpc.swmed.edu/bicf_nanocourses/ngs_analysis/fall2020)
### October 5th, 12th, 19th and 26th 2020

This 4-day intensive workshop will help researchers develop skills for managing, visualizing, analyzing, and integrating various types of omics data in their studies.  This course will cover:
- Short Read Alignments, Reference Genomes and Sequencing Technologies.
- Using Point and Click NGS Analysis workflows
- Gene Expression (RNASeq)
- Gene Regulation (ChIPSeq and ATACSeq).
- Data Integration in Gene Regulation and Expression
- Variation Detection Tools
- Somatic and Germline Variant Priorization in Cancer, Medelian and Complex Diseases

* * *
## [Accessing Public Data](https://bicf.pages.biohpc.swmed.edu/bicf_nanocourses/accessing_public_data)
### February 14th, 2020 (One day only)
9am to 5pm; E2.305 (Library conference room, will require badge entry)

For many types of sequencing analyses, we need access to public data stored in various databases and repositories. This workshop will discuss types of genomic reference data available through public databases such as Ensembl, NCBI, UCSC, ENCODE, and TCGA and step through how to find and download this data. The workshop will also explore how to find and download publicly available experimental data, such as data (FASTQ files and count matrices) from published papers, using GEO and the SRA repositories. While most of the workshop will access data using a web browser, downloading data from the SRA will require beginner knowledge of the command-line interface and TCGA
analysis will require beginner R.

Training Topics:

- TCGA Data Access
- ENCODE and the UCSC Browswer

Course Size: 15 students

Academic Credit: Not available

* * *
## [Introduction to R for Beginners, Level I and II](https://bicf.pages.biohpc.swmed.edu/bicf_nanocourses/data_science)
### January 10th, 17th, 24th, 31st, 2020

- Jan. 10, 17; 9am to 5pm; NB2.100A
- Jan. 24; 8am-3pm; NB2.100A
- Jan. 31; 8am-5pm; NG3.202 (Note: different room)

Do you want to be to do simple statistical analyses yourself? Do you find yourself spending time and effort generating the same plots and statistics for each project? <a>R</a> is a freely available language and programming environment for statistical computing and graphics which provides a wide variety of statistical and graphical techniques: linear and nonlinear modeling, statistical tests, time series analysis, classification, clustering, etc.

You will need to bring a laptop computer with the latest version of R-Studio installed. If you already have R-studio, please make sure you have the latest version of R installed.

Training Topics:

- Introduction to R
- Importing and Cleaning Data with Tidyverse
- Manipulating and Merging Data
- Data Visualization with GG Plot
- Statistical Tests
- Linear Modeling
- Scripting and Markdown
- Bioconductor and Public data (TCGA)

Course Size: 15 students

Academic Credit: 1 credit hour for each course level

* * *

## [Deep Learning for Healthcare Genomics](https://bicf.pages.biohpc.swmed.edu/bicf_nanocourses/markiiisys)

This workshop teaches you how to apply deep learning to detect chromosome co-deletion and search for motifs in genomic sequences.

You’ll learn how to:
- Understand the basics of convolutional neural networks (CNNs) and how they work
- Apply CNNs to MRI scans of low-grade gliomas (LGGs) to determine 1p/19q chromosome co-deletion status
- Use the DragoNN toolkit to simulate genomic data and to search for motifs
- Upon completion, you’ll be able to: understand how CNNs work, evaluate MRI images using CNNs, and use real regulatory genomic data to research new motifs.

* * *
## [Computational Image Analysis](https://bicf.pages.biohpc.swmed.edu/bicf_nanocourses/image_analysis)

This nanocourse offers an introduction to state-of-the-art computer vision methods to convert image data into quantitative information. The four-day intensive course covers image analysis fundamentals using theory lectures and hands-on computer exercises using popular image analysis programs such as ImageJ, CellProfiler and Matlab. Biomedical scientists will gain the background to (1) search for and evaluate existing image analysis software, and (2) start devising their own image analysis pipeline/software. The course will also include and "image analysis therapy" session where the class can brainstorm about each other's image analysis problems.

The course is open to any interested person at UTSW, provided they utilize imaging and are interested in computational image analysis for their research. Some background in mathematics and programming is a plus, e.g., completion of the Mathematical Foundations of Quantitative Biology course and the Matlab bootcamp.

Time: This is a four-day course that runs from 9 a.m. to 5 p.m. each day

Class Location:

Training Topics:

- Image Enhancement & Filtering
- Segmentation
- Object Detection & Tracking
- Colocalization
- Morphological Operators
- Machine Learning Approaches


This course is also part of the [Computational and Systems Biology curriculum](http://www.utsouthwestern.edu/education/graduate-school/programs/phd-degrees/specialized--supplemental-phd/computational-and-systems-biology/course-descriptions.html)

* * *

## [Python II](https://bicf.pages.biohpc.swmed.edu/bicf_nanocourses/python_2)

This nanocourse builds upon the progress of Python I and continues to develop capabilities for for scientific computing and advanced data analysis, including basic machine learning.

Python is an open-source, fun, easy to learn, and powerful programming language.
With deep community support and wide ranging deployment across many domains, Python is a worthy tool for projects large and small that any computational scientist should keep on hand.

Topics for this two day (all-day) course will include:

- Numerical arrays/matrices (numpy/scipy)
- Data structures/DataFrames (pandas)
- Plotting and Visualizing data (matplotlib, seaborn, bokeh, scikit-learn)
- Reading and Writing Data Files

* * *

## [Machine Learning I](https://bicf.pages.biohpc.swmed.edu/bicf_nanocourses/ML_1)

Are you interested in machine learning?  This course is an introductory course for students to learn the basics.  Programming experience in Python is mandatory.


* * *

## [NCBI Workshops presented by NIH](https://bicf.pages.biohpc.swmed.edu/bicf_nanocourses/ncbi_nanocourse/)

NCBI offers a series of [modular workshops](https://www.ncbi.nlm.nih.gov/education/workshops/) on related set of NCBI resources. Each module consists of a 3-hour largely hands-on session emphasizing a different set of NCBI resources specifically designed for the UTSW community.  Please bring your laptop computer.

Topics will include:
- NCBI Tools for NIH Grantees and those wanting to be Grantees
- NCBI Resources for Human Clinical/Phenotype Variation Research
- NCBI Resources for Clinical Microbiology Research
- NCBI Tools for NIH Grantees and those wanting to be Grantees

* * *

## [Introduction to Python I](https://bicf.pages.biohpc.swmed.edu/bicf_nanocourses/python_1)

This nanocourse will introduce Python for scientific computing.  Python is an open-source, fun, easy to learn, and powerful programming language.
With deep community support and wide ranging deployment across many domains, Python is a worthy tool for projects large and small that any computational scientist should keep on hand.

Topics for this two day (all-day) course will include:

- Basic install, setup, and IDEs
- Basic Syntax
- Conditional statements, loops, functions
- Modules, classes, scripting, debugging
- Numerical arrays/matrices (numpy/scipy)
- Data structures (pandas)

The course will be interactive, with lectures followed by hands-on learning and exercises.
No previous programming experience is necessary. Familiarity with basic programming/scripting concepts is helpful.

* * *
